analyses/survival-analysis/02-summary-statics-TP53.Rmd

```diff
@@ -0,0 +1,315 @@
+                                age_at_diagnosis_year > 26 & 
+                                  age_at_diagnosis_year <= 40 ~ "(26,40]",
+                                age_at_diagnosis_year > 40 & 
+                                  age_at_diagnosis_year <= 60 ~ "(40,62]", 
+                                TRUE ~ "not reported")) %>%
+  distinct() %>% 
+  write_tsv(file.path(file.path(analyses_dir, "input", "HGG_survial_tp53.tsv")))
+
+## GBM samples
+GBM_df <- hist %>%
+    filter(grepl("^C3", sample_id)) %>%
+    left_join(tp53_score, by = "sample_id") %>%
+    ## transfer CPTAC samples age to column age_at_diagnosis_days
+    mutate(age_at_diagnosis_days = (HARMONY_age * 365.25),
+           age_at_diagnosis_year = HARMONY_age, 
+           reported_gender = HARMONY_Gender, 
+           molecular_subtype = "GBM", 
+           binned_age = HARMONY_age_class_derived, 
+           broad_mol = "GBM", 
+           mol_subtype_without_tp53 = "GBM") %>% 
+    select(sample_id, tp53_score, tp53_altered, 
+         OS_days, OS_status, EFS_days, EFS_event_type, 
+         reported_gender, race, age_at_diagnosis_days, age_at_diagnosis_year,
+         molecular_subtype, binned_age, binned_tp53, broad_mol, mol_subtype_without_tp53, status) %>% 
+    mutate(binned_age = case_when(age_at_diagnosis_year > 62 ~ ">62", TRUE ~ binned_age)) %>%
+    distinct() %>% 
+    write_tsv(file.path(file.path(analyses_dir, "input", "GBM_survial_tp53.tsv")))
+
+combined <- bind_rows(GBM_df, HGG_survival)
+
+```
+### test if there is a correlation between age and tp53 score
+
+```{r}
+## function to print the formula of linear regression 
+lm_eqn <- function(df, x, y){
+    m <- lm(y ~ x, df);
+    eq <- substitute(italic(y) == a + b %.% italic(x)*","~~italic(r)^2~"="~r2, 
+         list(a = format(unname(coef(m)[1]), digits = 2),
+              b = format(unname(coef(m)[2]), digits = 2),
+             r2 = format(summary(m)$r.squared, digits = 3)))
+    as.character(as.expression(eq));
+}
+
+ggplot(combined, aes(x = age_at_diagnosis_days, y = tp53_score)) + 
+  geom_smooth(method = "lm") + 
+  geom_point() + 
+  geom_text(y = 0.05, x = 20000, 
+            label = lm_eqn(combined, x= combined$age_at_diagnosis_days, y = combined$tp53_score), 
+            parse = TRUE)
+```
+
+### The distribution of TP53 score between age range and tp53 score 
+
+```{r}
+group_summary <- combined %>%
+  group_by(binned_age) %>%
+  summarize(count = n())
+
+# Filter the groups with at least 3 samples
+combined_filtered <- combined %>%
+  filter(binned_age %in% group_summary$binned_age[group_summary$count >= 3])
+
+ggplot(combined_filtered, aes(x = factor(binned_age, levels = c("[0,15]", "(15,26]", "(26,40]", "(40,62]", ">62", "not reported")), y = tp53_score)) + 
+  geom_boxplot() +
+  geom_jitter(aes(color = broad_mol), width = 0.3, alpha = 0.5) + 
+  stat_compare_means(comparisons = list(c("[0,15]", "(15,26]"), c("[0,15]", "(26,40]"), c("[0,15]", "(40,62]"), c("[0,15]", ">62"))) + 
```

Comment:
can you add in explicitly the test you are using, or add in the script via comment which is default